packages/dashboard/e2e/tests/utils/grid.ts

```diff
@@ -0,0 +1,134 @@
+import { Locator, Page } from '@playwright/test';
+import last from 'lodash/last';
+
+import { DragPosition, dragAndDrop } from './dragAndDrop';
+import { center } from './mousePosition';
+import { BoundingBox, getBoundingBox } from './locator';
+
+export const GRID_SIZE = 10;
+
+const WidgetSelector = '[data-gesture=widget]';
+const SelectionSelector = '[data-gesture=selection]';
+const GridSelector = '#container';
+
+type ComponentLibrarySelector = 'kpi' | 'text';
+
+const WidgetSelectorMap: { [key in ComponentLibrarySelector]: string } = {
+  kpi: 'add KPI widget',
+  text: 'add Text widget',
+};
```

Comment:
Same comment as the const assertion below. Let TypeScript do the work for you. Objects and arrays can be turned into type literals with const assertions.

Reply:
Misread PR. Proper recommendation is to use `keyof typeof` to derive a type from the object.